addons/mail/models/mail_mail.py

```diff
@@ -331,6 +332,32 @@ def _prepare_outgoing_body(self):
             return ''
         return self.env['mail.render.mixin']._replace_local_links(self.body_html)
 
+    def _personalize_outgoing_body(self, body, partner=False):
+        """Return a modified body based on the recipient (partner).
+
+        Note that it must be called even if the recipient is not a partner
+        because in that case elements are removed from the body.
+
+        :param str body: body to personalize for the recipient
+        :param Model partner: recipient
+        """
+        if '/mail/unfollow' not in body:  # Avoid unnecessary query if the body must not be personalized
+            return body
+        self.ensure_one()
+        followers = self.mail_message_id.notification_ids.filtered(lambda notif: notif.follower_id).res_partner_id
```

Comment:
Mmmh ... probably part of the query count issue... even with the prefetching, the mail sending process is still highly sequential, meaning we regularly add queries that increase with number of records to create ... did you check additional queries, are they geometrically linked to the number of records to create/notify ?

Reply:
Arf did not see this was a computed field... fearing this is effectively the source of our issues. Why aren't we directly checking the source record followers ?